nixpkgs_review/review.py

```diff
@@ -312,12 +459,51 @@ def list_packages(path: str, system: str, check_meta: bool = False) -> List[Pack
     ]
     if check_meta:
         cmd.append("--meta")
+
+    use_cache = False
+    #use_cache = True
+    # with cache, package.position is wrong
+    # example:
+    # Package(pname='gstreamermm', position='/home/user/.cache/nixpkgs-review/rev-a9f5b7dbfe16c81a026946f2c9931479be31171d-42/nixpkgs/pkgs/development/libraries/gstreamer/gstreamermm/default.nix:36')
+    # but store path is ok
+
+    if use_cache:
+
+        import diskcache
+
+        # parse commit hash from path
+        head_rev = path.split("/")[-2].split("-")[1]
+        cache_key = f"{head_rev} {system} check_meta={check_meta}"
+        debug(f"review.py: list_packages: cache_key = {cache_key}")
+        cache_dir = "/home/user/.cache/nixpkgs-review/list_packages/" # TODO
+        debug(f"review.py: list_packages: cache_dir = {cache_dir}")
+
+        cache_size_limit = 200 * 1000 * 1000, # 200MB (default: 1GB)
+
+        cache = diskcache.Cache(cache_dir, size_limit = cache_size_limit)
```

Comment:
I would prefer to have no dependencies outside the python stdlib in `nixpkgs-review`.

Reply:
eval cache is now a "stupid file cache" with *.xml.gz files (todo: garbage collect)